packages/sources/ncfx/src/adapter.ts

```diff
@@ -44,35 +47,77 @@ interface Message {
   mid?: number
 }
 
+interface ForexMessage {
+  [pair: string]: { price: number; timestamp: string }
+}
+
+type Message = CryptoMessage | ForexMessage
+
+export type TOptions = {
+  from: string
+  to: string
+  inverse: boolean
+}
+
+interface Pair {
+  pair: string
+  inverse: boolean
+}
+
+const getIncludesOptions = (
+  //@ts-expect-error no-unused-vars
+  validator: Validator<TInputParameters>,
+  include: IncludePair,
+): TOptions | undefined => {
+  return {
+    from: include.from,
+    to: include.to,
+    inverse: include.inverse || false,
+  }
+}
+
+const getPair = (input: AdapterRequest): Pair => {
+  const validator = new Validator(
+    input,
+    endpoints.crypto.inputParameters,
+    {},
+    { includes, shouldThrowError: false },
+  )
+  if (validator.error) return { inverse: false, pair: '' }
+  const endpoint = input.data.endpoint
+
+  const { from, to, inverse } = util.getPairOptions<TOptions, TInputParameters>(
+    AdapterName,
+    validator,
+    getIncludesOptions,
+    (base: string, quote: string) => ({
+      from: base,
+      to: quote,
+      inverse: false,
+    }),
+  ) as TOptions
```

Comment:
There was a desire to split `getPairOptions` into separate methods to prevent this type coercion. Depending on whether this or that other change is merged in first, this might need to be removed.